src/parser/mod.rs

```diff
@@ -119,4 +119,75 @@ mod test {
 
         println!("{:?}", decls);
     }
+
+    #[test]
+    #[ignore]
+    fn test_parser_for_modular_exp() {
+        // Modular Exponentiation Example (Low to High Algorithm)
+        // https://en.wikipedia.org/wiki/Modular_exponentiation
+        // Example for 2^4
+        // 2^4 = 2^2 * 2^2 = 4^2 = 16
+        // -------------------------
+        // | a | e | acc |
+        // |---|---|-----|
+        // | 2 | 4 | 1   |
+        // | 4 | 2 | 1   |
+        // | 16| 1 | 1   |
+        // | 16| 0 | 16  |
+        // -------------------------
+        // Example for 3^5
+        // 3^5 = 3^2 * 3^2 * 3 = 9^2 * 3 = 81 * 3 = 243
+        // -------------------------
+        // | a | e | acc |
+        // |---|---|-----|
+        // | 3 | 5 | 1   |
+        // | 9 | 2 | 3   |
+        // | 81| 1 | 3   |
+        // | 81| 0 | 243 |
+        // -------------------------
+
+        let circuit = "
+        machine modular_exp(signal a, signal e) (signal acc: field) {
+            // a, e and acc are created automatically as shared signals
+
+            state initial {
+                acc <== 1;
+                -> middle {
+                    acc', a', e' <== acc, a, e;
+                }
+            }
+
+            state middle {
+                signal div;
+                signal new_acc;
+
+                div <-- e / 2;
+                div * 2 === e;
```

Comment:
@rute aren't these both constraitns contradictory?

div * 2 === e;
div * 2 + rem === e;

Reply:
Following the circom operators https://docs.circom.io/circom-language/basic-operators/#arithmetic-operators

We should use `\` for this division